### PR TITLE
Support  of Keyword WECON. 

### PR DESCRIPTION
**Pull request by GitPaean**
To handle the automatic shut-in related, added the flag to indicate automatic shut-in to the Well class.

### CODE REVIEW

**Comment by joakim-hove** [comment]
This looks good - test and all! Will review more thoroughly tomorrow; only minor comment so far is that `WellEconProductionLimits` is *really* long - what about `WellEconLimits`?

**Comment by GitPaean** [comment]
Yes. We can do that, though it is really only applied to production wells. I will address that with other comments together tomorrow.

File: opm/parser/eclipse/EclipseState/Schedule/Well.hpp [hunk 1]
```diff
@@ -105,6 +107,9 @@ namespace Opm {
         bool                           setSolventFraction(size_t timeStep , const double fraction);
         const double&                  getSolventFraction(size_t timeStep) const;
 
+        bool                            setEconProductionLimits(const size_t timeStep, const WellEconProductionLimits& properteis);
```
Comment (pgdr) [inline opm/parser/eclipse/EclipseState/Schedule/Well.hpp]: properties*

File: opm/parser/eclipse/EclipseState/Schedule/Well.hpp [hunk 2]
```diff
@@ -45,7 +46,8 @@ namespace Opm {
     class Well {
     public:
         Well(const std::string& name, std::shared_ptr<const EclipseGrid> grid , int headI, int headJ, Value<double> refDepth , Phase::PhaseEnum preferredPhase,
-             std::shared_ptr< const TimeMap > timeMap, size_t creationTimeStep, WellCompletion::CompletionOrderEnum completionOrdering = WellCompletion::TRACK, bool allowCrossFlow = true);
+             std::shared_ptr< const TimeMap > timeMap, size_t creationTimeStep, WellCompletion::CompletionOrderEnum completionOrdering = WellCompletion::TRACK,
```
Comment (pgdr) [inline opm/parser/eclipse/EclipseState/Schedule/Well.hpp]: I don't have strong opinions on this, but I think 160 character lines is a bit over the top.  Could you perhaps break these lines?

File: testdata/integration_tests/SCHEDULE/SCHEDULE_WECON [hunk 3]
```diff
@@ -0,0 +1,44 @@
+START
+  10 MAI 2007   /
```
Comment (pgdr) [inline testdata/integration_tests/SCHEDULE/SCHEDULE_WECON]: MAY?

**Comment by pgdr** [comment]
It looks good!

**Comment by joakim-hove** [comment]
I am not saying you *should* do this; but have you considered extending the [`WellProductionProperties`](https://github.com/OPM/opm-parser/blob/master/opm/parser/eclipse/EclipseState/Schedule/WellProductionProperties.cpp) structure instead of adding a completely new structure?

**Comment by GitPaean** [comment]
> I am not saying you should do this; but have you considered extending the WellProductionProperties structure instead of adding a completely new structure?

Yes. I thought about that. `WellProductionProperties` is mostly about the well control for the production wells, and it will enter the well equations (maybe not everything).  `WECON` is really about whether the running of wells still economical. We shut/close some wells or connections when the output of the well is not good enough anymore after running for each report step. They are kind of independent of the purpose and usage. But for sure, they are all about the production wells. We can possibly put the whole `WellEconLimits` as a whole in the `WellProductionProperties`, while it may not bring much benefit. Implementing the `WECON` on top of `WellProductionProperties` will make the `WellProductionProperties` much longer and more complicated, i do not think it is a good idea.

File: testdata/integration_tests/SCHEDULE/SCHEDULE_WECON (hunk 3 above)
Comment (joakim-hove) [inline testdata/integration_tests/SCHEDULE/SCHEDULE_WECON]: Eclipse seems to prefer 'MAI'?!

**Comment by joakim-hove** [comment]
> i do not think it is a good idea.

Fair enough - it was just a question. I will merge when you addressed the micro issues raised in the review:

1.  [Long lines](https://github.com/OPM/opm-parser/pull/865#discussion_r68969691)
2.  [Spelling](https://github.com/OPM/opm-parser/pull/865#discussion_r68969463)
3. Optional: [rename] (https://github.com/OPM/opm-parser/pull/865#issuecomment-229382710)

File: testdata/integration_tests/SCHEDULE/SCHEDULE_WECON (hunk 3 above)
Comment (GitPaean) [inline testdata/integration_tests/SCHEDULE/SCHEDULE_WECON]: `testdata/integration_tests/SCHEDULE/` there are lot of decks using `MAI`. I am not sure how Eclipse and OPM handle this. From the manual, it should be `MAY`.

**Comment by GitPaean** [comment]
> Fair enough - it was just a question. I will merge when you addressed the micro issues raised in the review:

> Long lines
> Spelling
> Optional: rename

I broke the long lines and corrected the spelling. I chose to keep the name. To me, it is slightly better to remind that the keyword is for production well only.  Please let me know if you insist in changing it.

**Comment by joakim-hove** [comment]
In she goes!

**Pull request merged**